File: core/acp-packet/src/main/java/pers/acp/packet/iso8583/ISO8583FieldProperties.java
package pers.acp.packet.iso8583;

import pers.acp.core.config.base.BaseProperties;
import pers.acp.core.log.LogFactory;
import pers.acp.core.tools.CommonUtils;

/**
 * @author zhangbin by 2018-2-3 21:14
 * @since JDK 11
 */
public class ISO8583FieldProperties extends BaseProperties {

    private static final LogFactory log = LogFactory.getInstance(ISO8583FieldProperties.class);

    /**
     * 获取数据库配置实例
     *
     * @return 数据库配置实例
     */
    public static ISO8583FieldProperties getInstance() {
        try {
            String propertiesFileName = "/iso8583.properties";
            return (ISO8583FieldProperties) getInstance(ISO8583FieldProperties.class, propertiesFileName, CommonUtils.getWebRootAbsPath() + propertiesFileName);
        } catch (Exception e) {
            log.error(e.getMessage(), e);
            return null;
        }
    }

}

File: core/acp-packet/src/main/kotlin/pers/acp/packet/iso8583/Iso8583FieldProperties.kt
package pers.acp.packet.iso8583

import pers.acp.core.CommonTools
import pers.acp.core.base.BaseProperties
import pers.acp.core.log.LogFactory

/**
 * @author zhang by 11/07/2019
 * @since JDK 11
 */
class Iso8583FieldProperties : BaseProperties() {
    companion object {

        private val log = LogFactory.getInstance(Iso8583FieldProperties::class.java)

        /**
         * 获取数据库配置实例
         *
         * @return 数据库配置实例
         */
        @JvmStatic
        fun getInstance(): Iso8583FieldProperties? =
                try {
                    val propertiesFileName = "/iso8583.properties"
                    getInstance(Iso8583FieldProperties::class.java, propertiesFileName, CommonTools.getWebRootAbsPath() + propertiesFileName) as Iso8583FieldProperties
                } catch (e: Exception) {
                    log.error(e.message, e)
                    null
                }
    }

}
